src/sentry/integrations/jira/utils/create_issue_schema_transformers.py

```diff
@@ -0,0 +1,133 @@
+    JiraField,
+    JiraSchemaTypes,
+)
+from sentry.shared_integrations.exceptions import IntegrationFormError
+
+
+class JiraSchemaParseError(Exception):
+    pass
+
+
+def parse_number_field(num_str: Any) -> int | float:
+    try:
+        if isinstance(num_str, str) and "." in num_str:
+            return float(num_str)
+
+        return int(num_str)
+    except ValueError:
+        raise JiraSchemaParseError(f"Invalid number value provided for field: '{num_str}'")
+
+
+TransformerType = Mapping[str, Callable[[Any], Any]]
+
+T = TypeVar("T")
+
+
+def identity_transformer(input_val: T) -> T:
+    return input_val
+
+
+def id_obj_transformer(input_val: Any) -> dict[str, Any]:
+    return {"id": input_val}
+
+
+def get_type_transformer_mappings(user_id_field: str) -> TransformerType:
+    transformers = {
+        JiraSchemaTypes.user.value: lambda x: {user_id_field: x},
+        JiraSchemaTypes.issue_type.value: id_obj_transformer,
+        JiraSchemaTypes.option.value: lambda x: {"value": x},
+        JiraSchemaTypes.issue_link.value: lambda x: {"key": x},
+        JiraSchemaTypes.project.value: id_obj_transformer,
+        JiraSchemaTypes.number.value: parse_number_field,
+    }
+
+    return transformers
+
+
+def get_custom_field_transformer_mappings() -> TransformerType:
+    transformers = {
+        # JIRA_CUSTOM_FIELD_TYPES["select"]: identity_transformer,
+        JIRA_CUSTOM_FIELD_TYPES["tempo_account"]: parse_number_field,
+        JIRA_CUSTOM_FIELD_TYPES["sprint"]: parse_number_field,
+        # JIRA_CUSTOM_FIELD_TYPES["epic"]: identity_transformer,
+        JIRA_CUSTOM_FIELD_TYPES["rank"]: id_obj_transformer,
+    }
+
+    return transformers
+
+
+def get_transformer_for_field(
+    type_transformers: TransformerType, custom_transformers: TransformerType, jira_field: JiraField
+) -> Callable[[Any], Any]:
+    transformer = None
+    if jira_field.is_custom_field():
+        assert jira_field.schema.custom
+        transformer = custom_transformers.get(jira_field.schema.custom)
+
+    if not transformer:
+        field_type = jira_field.get_field_type()
+
+        if field_type:
+            transformer = type_transformers.get(field_type)
+
+    if not transformer:
+        transformer = identity_transformer
+
+    return transformer
+
+
+def transform_fields(
+    user_id_field: str, jira_fields: Iterable[JiraField], **data
+) -> Mapping[str, Any]:
+    transformed_data = {}
+
+    # Special handling for fields that don't map cleanly to the transformer logic
+    data["summary"] = data.get("title")
+    if labels := data.get("labels"):
+        data["labels"] = [label.strip() for label in labels.split(",") if label.strip()]
+
+    type_transformers = get_type_transformer_mappings(user_id_field)
+    custom_field_transformers = get_custom_field_transformer_mappings()
+
+    for field in jira_fields:
+        field_data = data.get(field.key)
+
+        # We don't have a mapping for this field, so it's probably extraneous
```

Comment:
will people complain if a field is not ported over when they try to create an issue?

Reply:
Probably! But this seems like a less terrible experience than mismapping a field, failing to create an issue, and not alerting them that something is broken. Might be worth collecting all of these unmapped fields and reporting them to the UI, though I'm not sure how spammy that will get.